src/qe/mbp/mbp_qel.cpp

```diff
@@ -178,7 +178,7 @@ class mbp_qel::impl {
 
         std::function<bool(expr *)> non_core = [&](expr *e) {
             if (is_app(e) && is_partial_eq(to_app(e))) return true;
-            if (m.is_ite(e)) return true;
+            if (m.is_ite(e) || m.is_or(e) || m.is_implies(e) || m.is_and(e) || m.is_distinct(e)) return true;
```

Comment:
is_xor? is_iff? (is_iff is shorthand for is_eq and arguments are Boolean). is_distinct also takes non-boolean arguments so not sure what the discriminator is here.